### DISCUSSION TITLE
Framework Init and Updating

### DISCUSSION THREAD

**Discussion by wchorski** [keystonejs/keystone#9225] [Questions]
I have been building with [framework-nextjs-app-directory](https://github.com/keystonejs/keystone/tree/main/examples/framework-nextjs-app-directory) and was looking to upgrade it's packages, inline with Keystone's framework.

I see now they use `pnpm` to manage packages and a bit confused on the `workspace` and `catalog` namespaces inside the framework's `package.json`

## Before
Before there was no `pnpm`. The framework's `package.json` was easily portable over to a local directory where I could run `npm i`, remove stuff like `"@preconstruct/next": "^4.0.0",`, and start building

## How to initialize a framework?
How would I initialize a framework in my own repo where `framework-nextjs-app-directory` is the root directory? I understand how to work inside the `keystone` monorepo to build the app, but I would like to commit to a personal repo my changes. 

## How to update framework?
Also how would I go about update these packages when Keystone updates the framework?

**Comment by dcousens** [comment]
Interesting questions @wchorski, pnpm [`catalog:`](https://pnpm.io/catalogs) are a maintenance help for us as maintainers, but I can see how that might ruin the example for new users.

At this time I don't have any universal instructions for how to use the examples as starter templates, but maybe that could be an option for using the `create-keystone-app` package?  The configuration your repository really depends on your project.
